arcgis-ios-sdk-samples/Layers/Query related features from non-spatial table/QueryRelatedFeaturesNonSpatialTableViewController.swift

```diff
@@ -0,0 +1,157 @@
+            guard let self = self else { return }
+            if let comments = result?.featureEnumerator().allObjects {
+                // Show the records from the service request comments table in the list view control.
+                self.commentsArray = comments
+                // Enable the button after the map and features have been loaded.
+                self.queryBarButtonItem.isEnabled = true
+            } else if let error = error {
+                self.presentAlert(error: error)
+            }
+        }
+    }
+    
+    /// Query related features for the selected feature.
+    func queryCommentsTable(feature: AGSArcGISFeature) {
+        // Get the relationship that defines related service request features for features in the comments table (this is the first and only relationship).
+        guard let commentsRelationshipInfo = commentsTable?.layerInfo?.relationshipInfos.first else { return }
+        // Create query parameters to get the related service request for features in the comments table.
+        let relatedQueryParameters = AGSRelatedQueryParameters(relationshipInfo: commentsRelationshipInfo)
+        relatedQueryParameters.returnGeometry = true
+        // Query related features for the selected comment and its related query parameters.
+        commentsTable?.queryRelatedFeatures(for: feature, parameters: relatedQueryParameters) { results, error in
+            // Get the first related feature.
+            if let relatedFeature = results?.first?.featureEnumerator().nextObject() as? AGSArcGISFeature {
+                // Load the feature and get its geometry to show as a graphic on the map.
+                relatedFeature.load { [weak self] error in
+                    guard let self = self else { return }
+                    if let error = error {
+                        self.presentAlert(error: error)
+                    } else {
+                        // Get the feature's geometry.
+                        guard let serviceRequestPoint = relatedFeature.geometry as? AGSPoint else { return }
+                        // Create a graphic to add to the graphics overlay.
+                        let graphic = AGSGraphic(geometry: serviceRequestPoint, symbol: nil)
+                        self.selectedFeaturesOverlay.graphics.add(graphic)
+                        // Set the viewpoint to the the related feature.
+                        self.mapView.setViewpointCenter(serviceRequestPoint, scale: 150_000)
+                        // Enable the button after the feature has finished loading.
+                        self.queryBarButtonItem.isEnabled = true
+                    }
+                }
+            } else {
+                // Present an error message is the related feature is not found.
+                self.presentAlert(title: "Related feature not found. No Feature", message: nil)
```

Comment:
This contradicts Ting's [suggestion](https://github.com/Esri/arcgis-runtime-samples-ios/pull/1153#discussion_r836817254). @yo1995 I'm a little confused, can you elaborate?

Reply:
Oh my bad, I didn't notice on line 147 `self` is used to present alert. I thought the first completion block after `queryRelatedFeatures` didn't use self.

Reply:
The inner completion handler will capture `self` however it is in the current context. If it is weak, then it will capture it as weak. So as long as the outer completion handler captures it as weak and doesn't shadow `self` with a strong reference (`guard let self = self else { return }`), then the inner completion handler doesn't need `[weak self]`. Does that make sense?

Reply:
Yes, that all clears things up. Thank you both!